File: src/sum/sum.rb
# frozen_string_literal: true

# Sum :
class Sum
  def self.iterate(list)
    sum = 0
    list.each { |n| sum += n }
    sum
  end

  def self.recursive(list)
    return 0 if list.empty?

    list.first + recursive(list[1..])
  end

  def self.std(list)
    list.sum
  end
end

File: rust/src/sum/sum.rs
pub fn iterate(list: &[i32]) -> i32 {
    let mut sum = 0;
    for n in list {
        sum += n;
    }
    sum
}

pub fn recursive(list: &[i32]) -> i32 {
    if list.len() == 0 {
        return 0;
    }
    list.first().unwrap() + recursive(&list[1..])
}

pub fn std(list: &[i32]) -> i32 {
    list.iter().sum()
}
